public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  The `setValue` method checks the data object thoroughly before assigning any form control values.
+  It rejects a data object that doesn't match the `FormGroup` structure.
+  It rejects a data object if it is missing values for any control in the group.
+  It does return helpful errors.
+
+  Notice that you can _almost_ use the entire `hero` as the argument to `setValue`
+  because its shape is similar to the component's `FormGroup` structure.
+
+  You can only show the hero's first address and you must account for the possibility that the `hero` has no addresses at all.
+  This explains the conditional setting of the `address` property in the data object argument:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value-address')(format=".")
+
+:marked
+  ### _patchValue_
+  With **`patchValue`**, you can assign values to specific controls in a `FormGroup`
+  by supplying an object of key/value pairs for just the controls of interest. 
+
+  This example sets only the form's `name` control.
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'patch-value','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  With **`patchValue`** you have more flexibility to cope with wildly divergent data and form models.
+  But unlike `setValue`,  `patchValue` cannot check for missing control values and does not return errors.
+
+  ### When to set form model values (_ngOnChanges_)
+
+  Now you know _how_ to set the _form model_ values. But _when_ do you set them?
+  The answer depends upon when the component gets the _data model_ values.
+  
+  If the `HeroDetailComponent` is editing a _new_ hero, the data model consists entirely of default values
+  that you could copy to the `FormControls` when you create them in the component's constructor.
+  That's effectively what you did when you set each control to the empty string.
+
+  If the `HeroDetailComponent` is editing an _existing_ hero, you have to wait for that hero to arrive.
+  The component _might_ have the ability to retrieve the hero from the server (via an injected data service)
+  in which case you could set the form model in the [ngOnInit](lifecyle-hooks.html#oninit) lifecycle hook.
+  The final version of the `HeroDetailComponent` in the [Tour of Heroes](../tutorial/toh-pt6.html) does it this way.
+
+  The `HeroDetailComponent` in this reactive forms sample works a little differently.
+  It's the _detail_ view nested within a _master/detail_ `HeroListComponent` ([discussed below](#hero-list)).
+  
+  The `HeroListComponent` displays hero names to the user.
+  When the user clicks on a hero, the list component passes the selected hero into the `HeroDetailComponent`
+  by **binding to the latter's `hero` input property**.
+
++makeExample('reactive-forms/ts/app/hero-list.component.1.html', '','hero-list.component.html (simplified)')(format=".")
+
+:marked
+  In this approach, the value of `hero` in the `HeroDetailComponent` changes 
```

Comment:
@DeborahK Not following the question.  Yes, those symbols are needed. I searched the text and see that we didn't make a point of applying the decorators. They're in the code base but we don't say anything other than in our summary of imports on L#152.

I think our justification is that the reader is presumed to know what they are and do by the time they get to this guide. BUT maybe we shouldn't assume so much. Maybe we need to add a sense where we first add input/output fields. We'd link to the doc that discusses them more fully (not sure what that is).

What do you think @kapunahelewong ?